src/wallet.rs

```diff
@@ -51,6 +52,18 @@ impl Wallet {
         Ok(())
     }
 
+    pub fn new(name: String, xprv: ExtendedPrivateKey<SecretKey>, chain: Chain) -> Self {
```

Comment:
:question: Isn't deriving the `xprv` from a password and a salt something that we still wan to support? I think `waves` has got to be able to unlock a wallet using a simple password, rather than a full-blown `xprv`. What am I missing?

Reply:
I think you are correct. 
Here is how I think it should be changed:
- The `encryption_key` field needs to go on wallet. It kind of makes the password to unlock a wallet kind of useless and confusing.
- Derive the xprv from the name and password in the `new` fn. 
- Make a new `LockedWallet` or `EncryptedWallet` type.
- Implement serialisation on the `EncryptedWallet`